src/accounts.js

```diff
@@ -307,33 +353,71 @@ async function createKeyPair(password, salt, authAccountName, permissionName, op
 
   if (options.confirm) {
     const awaitTransactionOptions = getAwaitTransactionOptions(options);
-    await this.awaitTransaction(() => addPermission.bind(this)(authAccountName, [keys.publicKeys.active], permissionName, parentPermission, options), awaitTransactionOptions);
+    await this.awaitTransaction(() => addPermission.bind(this)(authAccountName, [keys.publicKeys.active], permissionName, parentPermission, false, options), awaitTransactionOptions);
   } else {
-    await addPermission.bind(this)(authAccountName, [keys.publicKeys.active], permissionName, parentPermission, options);
+    await addPermission.bind(this)(authAccountName, [keys.publicKeys.active], permissionName, parentPermission, false, options);
   }
 
   return keys;
 }
 
 async function createBridgeAccount(password, salt, authorizingAccount, options) {
-  options = {
-    oreAccountName: await generateAccountName.bind(this)(options.accountNamePrefix),
-    confirm: true,
-    ...options
-  };
+  let oreAccountName = null;
+  let isAccountUsable = false;
+  let transaction = null;
+  let transactionOptions;
+  let nameAlreadyExists = true;
 
-  let transaction;
+  const { confirm = true, oreAccountName: newAccountName } = options;
   const keys = await generateEncryptedKeys.bind(this)(password, salt, options.keys);
 
-  if (options.confirm) {
-    const awaitTransactionOptions = getAwaitTransactionOptions(options);
-    transaction = await this.awaitTransaction(() => this.createNewAccount(authorizingAccount, keys, options), awaitTransactionOptions);
-  } else {
-    transaction = await this.createNewAccount(authorizingAccount, keys, options);
+  if (!this.isNullOrEmpty(newAccountName)) {
+    nameAlreadyExists = await getNameAlreadyExists.bind(this)(newAccountName);
   }
 
+  if (!this.isNullOrEmpty(newAccountName)) {
+    // add the new active key to the newAccountName if the account name exists already on the chain with the active key set to unusedAccountPubKey
+    try {
+      isAccountUsable = await checkIfAccountNameUsable.bind(this)(newAccountName);
+      if (isAccountUsable) {
```

Comment:
might want to factor this function into a sub function that makes it more easy to read - its hard to read nested if statements and follow logic